### PR TITLE
Spectra v 2.7.8

### PR DESCRIPTION
**Pull request by adi3890**
- Improvement: Global - Enhanced the reusable block functionality for the following blocks: Google Maps, How To, Icon List, Inline Notice, Lottie, Marketing Button, Modal, Popup Builder, Post Grid and Post Carousel.
- Fix: Container - Resolved an issue in which the link tag would not work as intended on the front-end.
- Fix: How To - Resolved an issue in which the minimum minutes setting would be unskippable.
- Fix: Post Blocks - Resolved an issue in which the responsive styling would not work as intended in the editor.

### CODE REVIEW

**Review by rahulvarma722 — APPROVED** [review]

**Review by rajmohanreddy774 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -215,6 +215,12 @@ Our external packages use [Rating Star Component](https://github.com/n49/react-s
 
 ## Changelog ##
 
+### 2.7.8 - MONDAY, 4th SEPTEMBER 2023 ###
```
Comment (AbhijitNage123) [inline README.md:218]: Date should be TUESDAY, 5th SEPTEMBER 2023 in both Readme.

File: classes/class-uagb-block-helper.php [hunk 2]
```diff
@@ -1408,6 +1408,81 @@ public static function get_masonry_gallery_css() {
 
 			return $desktop . $tab_styling_css . $mob_styling_css;
 		}
+		/**
```
Comment (AbhijitNage123) [inline classes/class-uagb-block-helper.php:1411]: There should be one line of space after each function.

File: classes/class-uagb-block-helper.php [hunk 3]
```diff
@@ -1408,6 +1408,81 @@ public static function get_masonry_gallery_css() {
...
+				'yOverlayPosition'                => $attr[ 'yPositionOverlay' . $device_type ],
+				'yOverlayPositionType'            => $attr['yPositionOverlayType'],
+				'blendMode'                       => $attr['overlayBlendMode'],
+			);
+
+			$container_bg_css = self::uag_get_background_obj( $bg_obj, $overlay );
+
+			return $container_bg_css;
+		}
 		/**
```
Comment (AbhijitNage123) [inline classes/class-uagb-block-helper.php:1486]: Space needed after new function

File: README.md (hunk 1 above)
Comment (AbhijitNage123) [reply README.md:218]: Done here - https://github.com/brainstormforce/ultimate-addons-for-gutenberg/pull/3454

File: classes/class-uagb-block-helper.php (hunk 2 above)
Comment (AbhijitNage123) [reply classes/class-uagb-block-helper.php:1411]: Done here - https://github.com/brainstormforce/ultimate-addons-for-gutenberg/pull/3454

File: classes/class-uagb-block-helper.php (hunk 3 above)
Comment (AbhijitNage123) [reply classes/class-uagb-block-helper.php:1486]: Done here - https://github.com/brainstormforce/ultimate-addons-for-gutenberg/pull/3454

File: README.md (hunk 1 above)
Comment (adi3890) [reply README.md:218]: Thank you

**Review by AbhijitNage123 — APPROVED** [review]

File: blocks-config/uagb-controls/addInitialAttr.js [hunk 4]
```diff
@@ -95,27 +95,8 @@ const addInitialAttr = ( ChildComponent ) => {
...
-				'uagb/how-to',
-				'uagb/how-to-step',
-				'uagb/icon-list',
-				'uagb/icon-list-child',
 				'uagb/image-gallery',
-				'uagb/inline-notice',
-				'uagb/lottie',
-				'uagb/marketing-button',
-				'uagb/modal',
 				'uagb/popup-builder',
```
[the remark this answers is not in the document]
Comment (adi3890) [reply blocks-config/uagb-controls/addInitialAttr.js:99]: @rahulvarma722 can you please check for this.

File: includes/blocks/container/frontend.css.php [hunk 5]
```diff
@@ -38,52 +38,21 @@
...
-	'xOverlayPositionType'            => $attr['xPositionOverlayType'],
-	'yOverlayPosition'                => $attr['yPositionOverlayDesktop'],
-	'yOverlayPositionType'            => $attr['yPositionOverlayType'],
-	'blendMode'                       => $attr['overlayBlendMode'],
-);
-$container_bg_css_desktop = UAGB_Block_Helper::uag_get_background_obj( $bg_obj_desktop, 'no' );
-$container_bg_overlay_css = UAGB_Block_Helper::uag_get_background_obj( $bg_obj_desktop, 'yes' );
+$container_bg_css_desktop = UAGB_Block_Helper::get_background_css_by_device( $attr, 'Desktop', 'no' );
+$container_bg_css_tablet  = UAGB_Block_Helper::get_background_css_by_device( $attr, 'Tablet', 'no' );
+$container_bg_css_mobile  = UAGB_Block_Helper::get_background_css_by_device( $attr, 'Mobile', 'no' );
```
[the remark this answers is not in the document]
Comment (adi3890) [reply includes/blocks/container/frontend.css.php:43]: @rajmohanreddy774 as discussed before we can remove this.

[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply includes/blocks/container/frontend.css.php:43]: Removed in PR #3455

[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply includes/blocks/container/frontend.css.php:43]: Done in PR #3455

[the remark this answers is not in the document]
Comment (bsf-zanev) [reply includes/blocks/container/frontend.css.php]: Resolved in [this merge](https://github.com/brainstormforce/ultimate-addons-for-gutenberg/pull/3453#:~:text=Verified-,e8e6c8b,-Merge%20state).

[the remark this answers is not in the document]
Comment (bsf-zanev) [reply includes/blocks/container/frontend.css.php]: Resolved in [this merge](https://github.com/brainstormforce/ultimate-addons-for-gutenberg/pull/3453#:~:text=Verified-,e8e6c8b,-Merge%20state).

File: blocks-config/uagb-controls/addInitialAttr.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (rahulvarma722) [reply blocks-config/uagb-controls/addInitialAttr.js:99]: Yes we can remove

File: blocks-config/uagb-controls/generateCSSUnit.js [hunk 6]
```diff
@@ -1,4 +1,4 @@
-function generateCSSUnit( value, unit ) {
```
Comment (rahulvarma722) [inline blocks-config/uagb-controls/generateCSSUnit.js:1]: Is this change required

Comment (rajmohanreddy774) [reply blocks-config/uagb-controls/generateCSSUnit.js:1]: Yes it is required.

**Review by rahulvarma722 — APPROVED** [review]

**Review by bsf-zanev — APPROVED** [review]
Looks good to me.
